### PR TITLE
Low pop food events

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Fixed BluespaceCargoRule to be a single system, made it much more simple for adding new events based on it.
Added new 2 events for under 20 players:
Spawn 1-3 crates of McCargo (1-9 food boxes) on random ship.
Spawn a damaged McDelivery with 3 crates (3-9 food boxes)  in it, can happen 2 times in total.

## Why / Balance
Food for low pop in the case no one making any.

## How to test
addgamerule BluespaceMcCargo when you got a ship to get the random crates spawn
addgamerule BluespaceMcEvent to get the event to spawn

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame.

image

## Breaking changes
N/A

**Changelog**
:cl: dvir01
- add: Low pop of 20 and less might get free food from 2 new events now.

### CODE REVIEW

File: Resources/Prototypes/Catalog/Fills/Crates/food.yml [hunk 1]
```diff
@@ -157,3 +157,15 @@
...
+  parent: CrateFreezerMcCargo
+  suffix: Filled
+  components:
+  - type: StorageFill
+    contents:
+    - id: HappyHonkCargoFood
+    - id: HappyHonkCargoFood
+      prob: 0.5
+    - id: HappyHonkCargoFood
+      prob: 0.5
```
Comment (whatston3) [inline Resources/Prototypes/Catalog/Fills/Crates/food.yml:171]: Should be moved under an _NF folder.

**Review by whatston3 — CHANGES_REQUESTED** [review]
Left a bunch of suggestions.  I've made them all on a branch on my repo ([whatston3:2024-07-19-FoodEvent-suggestions](https://github.com/whatston3/frontier-station-14/tree/2024-07-19-FoodEvent-suggestions)).  Event as-is seems generous for low-pop, but eh.

I did rename the Minimum/Maximum variable, I added snack food (chips, raisins, chocolate) to replace the onion rings and put them in the side group with the fries, and I set the maxOccurrences for the BluespaceMcCargo event to 3.

Not sure if you're a fan of the orGroup & amount solution for the BluespaceMcCargo distribution.  I did like the normalized prob settings, those were nice.

Something I didn't do - would it make sense if the events had different names?  `BluespaceMcCargoCrate` vs. `BluespaceBrokenMcDelivery`?

Works well, happy to approve when the suggestions are reviewed.

File: Resources/Textures/_NF/Structures/Storage/Crates/mccargo.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by EmoGarbage404 (github) and modified by Potato1234x (github) for Space Station 14.",
```
[the remark this answers is not in the document]
Comment (dvir001) [reply Resources/Textures/_NF/Structures/Storage/Crates/mccargo.rsi/meta.json:4]: I forgor to fix

File: Resources/Prototypes/_NF/Events/events_nf.yml [hunk 3]
```diff
@@ -14,21 +14,44 @@
...
+  components:
+  - type: StationEvent
+    startAnnouncement: bluespace-cargo-event-announcement
+    weight: 5
+    duration: 35
+    maximumPlayers: 20
+  - type: BluespaceCargoRule
+    spawnerPrototype: CrateFoodMcCargo
+    flashPrototype: EffectFlashMcBluespace
+    requireSafeAtmosphere: true
```
[the remark this answers is not in the document]
Comment (dvir001) [reply Resources/Prototypes/_NF/Events/events_nf.yml:39]: Low pop, let it be.

**Comment by dvir001** [comment]
Mc Fixing time

File: Resources/Prototypes/_NF/Entities/Objects/Consumable/Food/Containers/box.yml [hunk 4]
```diff
@@ -11,14 +11,143 @@
...
+      prob: 0.25
+    - id: FoodCondimentPacketHotsauce
+      orGroup: CondimentPool
+      prob: 0.25
+    - id: FoodCondimentPacketKetchup
+      orGroup: CondimentPool
+      prob: 0.25
+    # Others
+    - id: FoodOnionRings
+    - id: FoodCondimentPacketPepper
```
[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Objects/Consumable/Food/Containers/box.yml]: > random fruit (grapes, apple, pear, orange).

Healthy food-stuffs? In my McCargo meal?! I want my money back.

**Review by ErhardSteinhauer — APPROVED** [review]
Do be working

**Review by whatston3 — APPROVED** [review]
Set the max occurrences for the mccargocrate event to 4, should be generous enough for low-pop conditions.  Happy with this, should be a good feature. 😃

**Pull request merged**